test/Microsoft.CodeAnalysis.Razor.Workspaces.Test/ProjectSystem/DefaultProjectSnapshotManagerTest.cs

```diff
@@ -10,289 +10,317 @@ namespace Microsoft.CodeAnalysis.Razor.ProjectSystem
+            projectManager.ProjectRemoved(project);
 
             // Assert
-            Assert.Empty(ProjectManager.Projects);
+            Assert.Empty(projectManager.Projects);
 
-            Assert.True(ProjectManager.ListenersNotified);
-            Assert.False(ProjectManager.WorkerStarted);
+            Assert.True(projectManager.ListenersNotified);
+            Assert.False(projectManager.WorkerStarted);
         }
 
         [Fact]
         public void ProjectRemoved_IgnoresUnknownProject()
         {
             // Arrange
-            var project = Workspace.CurrentSolution.AddProject("Test", "Test", LanguageNames.CSharp);
+            var testPieces = CreateProjectManagerPieces();
+            var project = testPieces.Project;
+            var projectManager = testPieces.ProjectManager;
 
             // Act
-            ProjectManager.ProjectRemoved(project);
+            projectManager.ProjectRemoved(project);
 
             // Assert
-            Assert.Empty(ProjectManager.Projects);
+            Assert.Empty(projectManager.Projects);
 
-            Assert.False(ProjectManager.ListenersNotified);
-            Assert.False(ProjectManager.WorkerStarted);
+            Assert.False(projectManager.ListenersNotified);
+            Assert.False(projectManager.WorkerStarted);
         }
 
         [Fact]
         public void ProjectsCleared_RemovesProject_NotifiesListeners_DoesNotStartBackgroundWorker()
         {
             // Arrange
-            var project = Workspace.CurrentSolution.AddProject("Test", "Test", LanguageNames.CSharp);
+            var testPieces = CreateProjectManagerPieces();
+            var project = testPieces.Project;
+            var projectManager = testPieces.ProjectManager;
 
-            ProjectManager.ProjectAdded(project);
-            ProjectManager.Reset();
+            projectManager.ProjectAdded(project);
+            projectManager.Reset();
 
             // Act
-            ProjectManager.ProjectsCleared();
+            projectManager.ProjectsCleared();
 
             // Assert
-            Assert.Empty(ProjectManager.Projects);
+            Assert.Empty(projectManager.Projects);
+
+            Assert.True(projectManager.ListenersNotified);
+            Assert.False(projectManager.WorkerStarted);
+        }
+
+        private static (Workspace Workspace, Project Project, TestProjectSnapshotManager ProjectManager) CreateProjectManagerPieces()
+        {
+            Project project = null;
+            var workspace = TestWorkspace.Create(ws =>
+            {
+                project = ws.CurrentSolution.AddProject("Test", "Test", LanguageNames.CSharp);
+            });
+            var projectManager = new TestProjectSnapshotManager(Enumerable.Empty<ProjectSnapshotChangeTrigger>(), workspace);
 
-            Assert.True(ProjectManager.ListenersNotified);
-            Assert.False(ProjectManager.WorkerStarted);
+            return (workspace, project, projectManager);
```

Comment:
As an aside, I've come to really like the pattern of just setting properties on the class.

Reply:
BTW you'll now have to skip ALL of these tests on net46 because of the issue with value tuples that will cause it to fail on appveyor. Workaround is to not use valuetuple

Reply:
> As an aside, I've come to really like the pattern of just setting properties on the class.

Ya me too. I forgot why I had to move away from that here. I'll have to look at it again.

Reply:
But yeah, that tuple thing is really painful right now. I didn't look but I suspect that's why your appveyor is teh failz0r